flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/rules/FlinkBatchRuleSets.scala

```diff
@@ -119,7 +119,9 @@ object FlinkBatchRuleSets {
         new CoerceInputsRule(classOf[LogicalMinus], false),
         ConvertToNotInOrInRule.INSTANCE,
         // optimize limit 0
-        FlinkLimit0RemoveRule.INSTANCE
+        FlinkLimit0RemoveRule.INSTANCE,
+        // unnest rule
+        LogicalUnnestRule.INSTANCE
```

Comment:
It seems that `LogicalUnnestRule` is also part of de-correlate. Why not put it in the `DECORRELATE` phase?

Reply:
`decorrelate` replaces all correlated expressions in RelNode with non-correlated expressions.
while `LogicalUnnestRule` rewrites `LogicalCorrelate` on `Uncollect` pattern to `LogicalCorrelate` on `LogicalTableFunctionScan` with `explode`, which will be converted to physical Correlate in physical optimization phase. so `LogicalUnnestRule` should be put in `DEFAULT_REWRITE`.

Reply:
Can you add these replies to code comments?